### PR TITLE
UHM-3511: Checkbox widget should be able to submit a specific value

### PR DESCRIPTION
**Pull request by jomadoye**

### CODE REVIEW

File: app/js/components/LabResultEntry.jsx [hunk 1]
```diff
@@ -135,15 +136,20 @@ export class LabResultEntry extends PureComponent {
...
                   "Stock Out",
                   "Blood Coagulation",
                   "Inappropriate Container"]}
                 widget="dropdown"
-                concept={CONSTANTS.labResultsDidNotPerformReason}
-                path="did-not-perform-dropdown"
+                concept={CONSTANTS.labResultsDidNotPerformQuestion}
+                path="did-not-perform-reason"
+                handleSelect={(field, selected) => {
+                  dispatch(change(formKey, `obs|path=did-not-perform-checkbox|concept=${CONSTANTS.labResultsDidNotPerformQuestion}`, selected));
```
Comment (mogoodrich) [inline app/js/components/LabResultEntry.jsx]: Why do we need to dispatch a change event in these cases?

File: app/js/components/LabResultEntry.jsx [hunk 2]
```diff
@@ -371,16 +383,25 @@ const mapStateToProps = (state) => {
...
-  const didNotPerformCheckbox = CONSTANTS && !!(selector(state, `obs|path=did-not-perform-checkbox|concept=${CONSTANTS.labResultsDidNotPerformQuestion}`));
+
+  const formKeyArr = Object.keys(form);
+  let didNotPerformCheckbox;
+  let formKey;
+  if (formKeyArr.length > 0) {
+    const selector = formValueSelector(formKeyArr[0]);
+    didNotPerformCheckbox = CONSTANTS && !!(selector(state, `obs|path=did-not-perform-checkbox|concept=${CONSTANTS.labResultsDidNotPerformQuestion}`));
+    [formKey] = formKeyArr;
+  }
```
Comment (mogoodrich) [inline app/js/components/LabResultEntry.jsx]: Can you explain what the above block of code is doing?  It's unclear to me.

File: app/js/components/LabResultEntry.jsx (hunk 1 above)
Comment (jomadoye) [reply app/js/components/LabResultEntry.jsx]: Each time the user toggles the dropdown, we update the value of the checkbox, to be same with the value of the dropdown.

File: app/js/components/LabResultEntry.jsx (hunk 2 above)
Comment (jomadoye) [reply app/js/components/LabResultEntry.jsx]: Yes, I noticed the redux-form generates a random/unique key each time, so I had to first get the key, that is `formKey`.
As demonstrated in the video, when the checkbox is checked the dropdown displays and when it's not checked the dropdown is hidden, I needed to keep track of the state of the checkbox for that, and that is what `didNotPerformCheckbox` is.

**Pull request merged**